arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -249,34 +234,56 @@ class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdateListen
      *
      * @since 100.6.0
      */
-    var originCamera: Camera = Camera(0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
+    var originCamera: Camera? = null
         set(value) {
             field = value
-            cameraController.originCamera = value
-            if (isTracking) {
-                resetTracking()
+            if (value != null) {
+                cameraController.originCamera = value
             }
         }
 
+    /**
+     * Property to determine if we are using only the first location provided by the LocationDataSource
+     *
+     * @since 100.6.0
+     */
+    private var useLocationDataSourceOnce: Boolean = true
+
     /**
      * This listener is added to every [LocationDataSource] used when using the [locationDataSource] property to receive
      * location updates.
      *
-     * Upon receiving an updated location, if we don't previously have an [initialLocation] we use that property to
+     * Upon receiving an updated location, if we don't previously have an [originCamera] we use the new location to
      * create a [Camera] and set that as the origin camera of the [cameraController]. On every other location update, if we
-     * are not using ARCore we set the current viewpoint camera of [sceneView] to a new Camera created from that location.
+     * are not using ARCore we create a new Camera using properties of the old camera. If user is using ARCore we reset
+     * the session to correct the tracking. We also reset the TransformationMatrix of the CameraController and stop the
+     * [locationDataSource] if [useLocationDataSourceOnce] is true.
      *
      * @since 100.6.0
      */
     private val locationChangedListener: LocationDataSource.LocationChangedListener =
         LocationDataSource.LocationChangedListener {
             it.location.position?.let { location ->
-                if (initialLocation == null) {
-                    initialLocation = location
-                    cameraController.originCamera = Camera(location, 0.0, 0.0, 0.0)
-                } else if (isUsingARCore != ARCoreUsage.YES) {
-                    val camera = sceneView.currentViewpointCamera.moveTo(location)
-                    sceneView.setViewpointCamera(camera)
+                // Always set the origin camera; then reset ARCore
+                val oldCamera = cameraController.originCamera
+
+                // Create a new camera based on our location and set it on the originCamera.
+                originCamera = if (originCamera == null) {
+                    Camera(location, 0.0, 90.0, 0.0)
+                } else {
+                    Camera(location, oldCamera.heading, oldCamera.pitch, oldCamera.roll)
```

Comment:
Should this only happen if we are not using ARCore? Based on the doc above...
```
On every other location update, if we are not using ARCore we create a new Camera using properties of the old camera.
```

Or does the doc need to be updated?